cdap-docs/examples-manual/source/examples/stream-conversion.rst

```diff
@@ -0,0 +1,163 @@
+
+    TimePartitionedFileSetArguments.setOutputPartitionTime(dsArguments, logicalTime);
+    TimePartitionedFileSet partitionedFileSet = context.getDataset("converted", dsArguments);
+    context.setOutput("converted", partitionedFileSet);
+
+- Note that the output file path is derived from the output partition time by the dataset itself::
+
+    LOG.info("Output location for new partition is: {}",
+             partitionedFileSet.getUnderlyingFileSet().getOutputLocation().toURI().toString());
+
+The Mapper itself is straight-forward: For each event, it emits an Avro record:
+
+.. literalinclude:: /../../../cdap-examples/StreamConversion/src/main/java/co/cask/cdap/examples/streamconversion/StreamConversionMapReduce.java
+     :language: java
+     :lines: 104-111
+
+In the ``afterSubmit`` method of the MapReduce program, if the run succeeded, the output file is
+registered as a new partition in the ``converted`` dataset. Note that this is due to a gap in the
+implementation of ``TimePartitionedFileSet``, which should add the partition in the output committer
+of its output format. This will be addressed in future CDAP releases:
+
+.. literalinclude:: /../../../cdap-examples/StreamConversion/src/main/java/co/cask/cdap/examples/streamconversion/StreamConversionMapReduce.java
+     :language: java
+     :lines: 85-95
+
+Building and Starting
+=====================
+
+- You can either build the example (as described `below
+  <#building-an-example-application>`__) or use the pre-built JAR file included in the CDAP SDK.
+- Start CDAP and deploy the application as described below in `Running CDAP Applications`_\ .
+- Once the application has been deployed and started, you can `run the example. <#running-the-example>`__
+
+Running CDAP Applications
+=========================
+
+.. include:: /../../developers-manual/source/getting-started/building-apps.rst
+    :start-line: 9
+
+Running the Example
+===================
+
+The ``StreamConversionWorkflow`` will run automatically every 5 minutes based on its schedule.
+To give it some data, you can use the provided script to send events to the stream, for example,
+to send 10000 events at a rate of roughly 2 per second::
+
+  bin/send-events.sh --events 10000 --delay 0.5
+
+You can now wait for the workflow to run, after which you can query the partitions in the
+``converted`` dataset, for example::
+
+  $ cdap-cli.sh execute \"show partitions cdap_user_converted\"
+  +============================================+
+  | partition: STRING                          |
+  +============================================+
+  | year=2015/month=0/day=28/hour=17/minute=30 |
+  | year=2015/month=0/day=28/hour=17/minute=35 |
+  | year=2015/month=0/day=28/hour=17/minute=40 |
+  +============================================+
+
+Note that in the Hive meta store, the partitions are registered with multiple dimensions rather
+than the time since the Epoch: the year, month, day of the month, hour and minute of the day.
```

Comment:
Yes, even though it is correct, it is the human readable form, and should get corrected. it will lead to too many mistakes otherwise.